src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,155 @@
+package duke.ui;
+
+import duke.command.Command;
+import duke.command.Commands;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.ToDoCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+
+
+/**
+ * Parses user input and returns Command.
+ *
+ * @author Lian Kok Hai
+ */
+public class Parser {
+    /**
+     * Parses user input and returns Command or throws
+     * DukeException when invalid input encountered.
+     *
+     * @param userInput User's string input.
+     * @return Command object to be executed.
+     * @throws DukeException
+     */
+    public Command parse(String userInput) throws DukeException {
+        Commands command = Parser.getCommand(userInput); // will catch any invalid command alrdy
+        Command c = null;
+        switch (command) {
+        case BYE:
+            c = new ExitCommand();
+            break;
+        case LIST:
+            c = new ListCommand();
+            break;
+        case MARK:
+            int index = Parser.parseMarkOrDeleteCommands(userInput);
+            c = new MarkCommand(index);
+            break;
+        case UNMARK:
+            index = Parser.parseMarkOrDeleteCommands(userInput);
+            c = new UnmarkCommand(index);
+            break;
+        case DELETE:
+            index = Parser.parseMarkOrDeleteCommands(userInput);
+            c = new DeleteCommand(index);
+            break;
+        case TODO:
+            c = new ToDoCommand(Parser.parseTodoCommand(userInput));
+            break;
+        case DEADLINE:
+            // parsed[0] is description of task; parsed[1] is by
+            String[] parsed = Parser.parseDeadlineCommand(userInput);
```

Comment:
Maybe you could consider naming parsed in a plural form as it is a multi-valued variable?